File: 0066-plus-one/0066-plus-one.java
import java.util.*;
class Solution {
    public int[] plusOne(int[] digits) {
        for(int i=digits.length -1; i> -1; i--){
            if(digits[i] != 9){
                digits[i] +=1;
                return digits;
            }else{
                digits[i] = 0;
            }
        }
        int[] ret = new int[digits.length + 1];
        Arrays.fill(ret,0);
        ret[0] = 1;
        return ret;
    }
}

File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, d: List[int]) -> List[int]:
        ans = []
        c=1
        for i in range(len(d)-1, -1, -1):
            t = d[i]+c
            c = t // 10
            ans.append(t % 10)
            
        
        if c==1:
            ans.append(1)
        return ans[::-1]
        
            